File: 1641-count-sorted-vowel-strings/1641-count-sorted-vowel-strings.cpp
class Solution {
public:
    int countVowelStrings(int n) {
    
        int endsWith[5] = { 1,1,1,1,1 };
        
        for (int i = 1; i < n; i++) {
            
            for (int idx = 1; idx < 5; idx++) {
                endsWith[idx] += endsWith[idx - 1];
            }
        }
        
        return endsWith[0] + endsWith[1] + endsWith[2] + endsWith[3] + endsWith[4];
    }
};

File: 1641-count-sorted-vowel-strings/1641-count-sorted-vowel-strings.py
class Solution:
    def countVowelStrings(self, n: int) -> int:
        
        endsWith = [1] * 6;
        endsWith[0] = 0
        
        for i in range(1, n):
            for idx in range(1, 6):
                endsWith[idx] += endsWith[idx - 1]
                
        return sum(endsWith)
                
            
        